crypto/objects/obj_dat.c

```diff
@@ -372,6 +372,10 @@ ASN1_OBJECT *OBJ_txt2obj(const char *s, int no_name)
         if (((nid = OBJ_sn2nid(s)) != NID_undef) ||
             ((nid = OBJ_ln2nid(s)) != NID_undef))
             return OBJ_nid2obj(nid);
+        if (*s < '0' || *s > '9') {
```

Comment:
Might `if (!ossl_isdigit(*s))` be safer since it is character set aware?

Reply:
Good point - done.